common/ast/src/parser/token.rs

```diff
@@ -0,0 +1,1117 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use logos::Logos;
+use logos::Span;
+
+pub use self::TokenKind::*;
+
+#[allow(non_camel_case_types)]
+#[derive(Logos, Clone, Copy, Debug, PartialEq)]
+pub enum TokenKind {
+    #[error]
+    #[regex(r"[ \t\n\f]+", logos::skip)]
+    Whitespace,
+
+    #[regex(r"--[^\t\n\f]*", logos::skip)]
+    Comment,
+
+    #[regex(r"/\*([^\*]|(\*[^/]))*\*/", logos::skip)]
+    CommentBlock,
+
+    #[regex(r#"[_a-zA-Z][_$a-zA-Z0-9]*"#)]
+    Ident,
```

Comment:
`Ident` should be distinguished with reserved words. For example:

```
SELECT select FROM t; -- not allowed
SELECT select_result FROM t; -- allowed
```

I think we need a complicated regular expression to handle this, or make an extra pass to check validity of `Ident`s(not a good choice).

How do you think about this?

Reply:
I think this should be the work for the parser, E.g, your first example will be lexed into `SELECT SELECT FROM ident(t)` on which the parser should be able to detect a grammar error.

Reply:
I was concerning about the ambiguation between reserved words and identifier, but I found `logos` do have a clear rule to handle this.

It looks good to me now.